testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliControllerConfigTestCase.java

```diff
@@ -0,0 +1,441 @@
+     * Protocol should be taken from default-protocol
+     */
+    @Test
+    public void testDefaultControllerNoExplicitPort() {
+        Protocol expectedProtocol = HTTPS_REMOTING;
+        JbossCliConfig jbossCliConfig = new JbossCliConfig()
+                .setUseLegacyOverride(true)
+                .setDefaultProtocol(expectedProtocol)
+                .setDefaultControllerProtocol(null)
+                .setDefaultControllerPort(INVALID_PORT);
+        jbossCliConfig.writeJbossCliConfig(tempJbossConfigFile);
+        CliProcessWrapper cli = getTestCliProcessWrapper(false);
+        try {
+            cli.executeInteractive();
+            boolean returnConnect = cli.pushLineAndWaitForResults("connect", DISCONNECTED_PROMPT);
+            assertTrue(returnConnect);
+            String output = cli.getOutput();
+            assertTrue(output.contains(":" + INVALID_PORT));
+            assertTrue(output.contains(expectedProtocol + "://"));
+        } catch (Exception ex) {
+            fail(ex.getLocalizedMessage());
+        } finally {
+            cli.destroyProcess();
+        }
+    }
+
+    /**
+     * Tests if default-protocol is overridden by protocol defined in default-controller
+     */
+    @Test
+    public void testDefaultProtocolOverriding() {
+        Protocol expectedProtocol = HTTPS_REMOTING;
+        JbossCliConfig jbossCliConfig = new JbossCliConfig()
+                .setUseLegacyOverride(true)
+                .setDefaultProtocol(REMOTE)
+                .setDefaultControllerProtocol(expectedProtocol)
+                .setDefaultControllerPort(INVALID_PORT);
+        jbossCliConfig.writeJbossCliConfig(tempJbossConfigFile);
+        CliProcessWrapper cli = getTestCliProcessWrapper(false);
+        try {
+            cli.executeInteractive();
+            boolean returnConnect = cli.pushLineAndWaitForResults("connect", DISCONNECTED_PROMPT);
+            assertTrue(returnConnect);
+            String output = cli.getOutput();
+            assertTrue(output.contains(":" + INVALID_PORT));
+            assertTrue(output.contains(expectedProtocol + "://"));
+        } catch (Exception ex) {
+            fail(ex.getLocalizedMessage());
+        } finally {
+            cli.destroyProcess();
+        }
+    }
+
+    /**
+     * Tests implicit CLI settings with empty config file,
+     * should attempt connection to remote+http://localhost:9990
+     */
+    @Test
+    public void testImplicitSettings() {
+        JbossCliConfig jbossCliConfig = new JbossCliConfig(); //empty config
+        jbossCliConfig.writeJbossCliConfig(tempJbossConfigFile);
+        CliProcessWrapper cli = getTestCliProcessWrapper(false);
+        try {
+            cli.executeInteractive();
+            cli.pushLineAndWaitForResults("connect");
+            boolean returnState = cli.pushLineAndWaitForResults(READ_SERVER_STATE, CONNECTED_PROMPT);
+            assertTrue(returnState);
```

Comment:
The following output is from our Jenkins node:
```
The controller is not available at localhost:9990: java.net.ConnectException: WFLYPRT0053: Could not connect to remote+http://localhost:9990. The connection failed: WFLYPRT0053: Could not connect to remote+http://localhost:9990. The connection failed: Connection refused
[disconnected /] :read-attribute(name=server-state)

```
I cannot reproduce this error locally. The CLI should be able to connect to the controller (remote+http://localhost:9990) using default values, no?